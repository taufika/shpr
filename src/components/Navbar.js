import React from 'react';

export default class Navbar extends React.Component {
  render() {
    return <h1>ulalalala</h1>;
  }
};
