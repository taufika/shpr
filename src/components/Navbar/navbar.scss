@import '../../variables/colors.scss';

.navbar {
  display: flex;
  height: 80px;
  background-color: #fff;
  justify-content: space-between;
  align-items: center;
  padding: 0px 16px;

  .logo {
    width: 120px;
  }

  .user-info {
    display: flex;
    align-items: center;

    .user-name {
      margin-left: 12px;
      color: $primary-color;
    }

    .user-avatar {
      width: 32px;
      height: 32px;
      overflow: hidden;
      margin-left: 16px;
      border-radius: 50%;
      position: relative;

      img {
        height: 100%;
      }
    }
  }
}