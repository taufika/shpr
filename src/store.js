import { configureStore } from '@reduxjs/toolkit'
import usersReducer from './features/users';
import sidebarReducer from './features/sidebar';

export default configureStore({
  reducer: {
    users: usersReducer,
    sidebar: sidebarReducer,
  },
})
