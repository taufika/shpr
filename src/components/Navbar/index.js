import React from 'react';
import { connect } from 'react-redux'
import Logo from '../../assets/shipper-logo-black.png';
import Avatar from '../../assets/default_avatar.jpg';

import { toggleSidebar } from '../../features/sidebar'

import './navbar.scss';

class Navbar extends React.Component {
  constructor(props) {
    super(props);
    this.toggleSidebar = this.toggleSidebar.bind(this);
  }
  toggleSidebar() {
    this.props.dispatch(toggleSidebar());
  }
  render() {
    return <nav className='navbar'>
      <button className='hamburger-menu' onClick={this.toggleSidebar} data-testid='navbar-hamburger'>
        <i className="fas fa-bars" />
      </button>
      <div className='logo'>
        <img src={Logo} alt='logo' className='logo' />
      </div>
      <section className='user-info'>
        <h4 className='hello'>Hello, 
          <span className='user-name'>{this.props.user.name || ''}</span>
        </h4>
        <div className='user-avatar'>
          <img src={Avatar} alt='avatar' />
        </div>
      </section>
    </nav>;
  }
};

export default connect()(Navbar);
