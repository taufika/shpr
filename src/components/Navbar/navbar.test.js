import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store';
import Navbar from '.';

const user = {
  name: 'Shipper User',
};

const wrapped = () => (
  <Provider store={store}><Navbar user={user} /></Provider>
);

describe('Component:Navbar', () => {
  it('Should render and match snapshot', () => {
    const { container } = render(wrapped());
    expect(container).toMatchSnapshot();
  });

  it('Should let user change sidebar active state using hamburger menu', () => {
    render(wrapped());
    expect(store.getState().sidebar.isSidebarActive).toBe(false);
    const hamburger = screen.getByTestId('navbar-hamburger');
    fireEvent.click(hamburger);
    expect(store.getState().sidebar.isSidebarActive).toBe(true);
  });
});