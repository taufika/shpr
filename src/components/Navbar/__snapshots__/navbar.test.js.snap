// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:Navbar Should render and match snapshot 1`] = `
<div>
  <nav
    class="navbar"
  >
    <button
      class="hamburger-menu"
      data-testid="navbar-hamburger"
    >
      <i
        class="fas fa-bars"
      />
    </button>
    <div
      class="logo"
    >
      <img
        alt="logo"
        class="logo"
        src="shipper-logo-black.png"
      />
    </div>
    <section
      class="user-info"
    >
      <h4
        class="hello"
      >
        Hello,
        <span
          class="user-name"
        >
          Shipper User
        </span>
      </h4>
      <div
        class="user-avatar"
      >
        <img
          alt="avatar"
          src="default_avatar.jpg"
        />
      </div>
    </section>
  </nav>
</div>
`;
